theano/tensor/tests/test_opt.py

```diff
@@ -3482,6 +3482,61 @@ def test_local_fill_useless():
     f(m_, x_)
 
 
+def test_local_elemwise_sub_zeros():
+    # Test opt local_elemwise_sub_zeros
+    # We test separetly for scalars, vectors and matrices
+    scalar = T.scalar()
+    vect = T.vector()
+    mat = T.matrix()
+
+    rng = numpy.random.RandomState(seed=utt.fetch_seed())
+    scalar_val = rng.rand(1)[0]
+    vect_val = rng.rand(5)
+    mat_val = rng.rand(3, 2)
+
+    mode = theano.compile.get_default_mode()\
+            .excluding('canonicalize', 'uncanonicalize',\
+                       'ShapeOpt', 'local_fill_to_alloc',\
+                       'local_elemwise_alloc')\
+            .including('local_elemwise_sub_zeros')
+
+    # Test scalar minus scalar
+    f = function([scalar], scalar-scalar, mode=mode)
+    # Check optimized graph is correct
+    assert isinstance(f.maker.fgraph.toposort()[0].op, T.Elemwise)
```

Comment:
Actually, there is still an Elemwise after the optimization. It's simply a fill op.

This is the graph before optimizing:
    Elemwise{sub,no_inplace} [id A] ''   
     |<TensorType(float64, scalar)> [id B]
     |<TensorType(float64, scalar)> [id B]

This is the graph after optimizing:
    Elemwise{second,no_inplace} [id A] ''   0
     |<TensorType(float64, scalar)> [id B]
     |TensorConstant{0.0} [id C]

I'll add a test to check specifically that the Elemwise is a fill op (Elemwise{second,no_inplace}).